test/performance/shmem_perf_suite/common.h

```diff
@@ -578,8 +573,8 @@ int streaming_node(perf_metrics_t my_info)
 static inline
 int target_node(perf_metrics_t my_info)
 {
-    return (my_info.my_node >= my_info.midpt &&
-        (my_info.my_node < (my_info.midpt + my_info.sztarget)));
+    return (my_info.my_node >= my_info.szinitiator &&
+        (my_info.my_node < (my_info.szinitiator + my_info.sztarget)));
```

Comment:
the reason midpt was compared was because we maintain the boundary between the half initiated for the initiator and the other half initiated for the target. Even when we decrease the amount on either side, we still maintain that boundary which defines initiator vs target

Reply:
Thanks, @kseager. The reason I changed this is that when `l/r` flag is used with a different number other than the `midpt`, this method will return a wrong answer for some of the processes that are actually target. For example, if we use `-l 3` with a total of 16 processes, then this method will provide the wrong answer for PE 3 to 7 with the previous implementation. Let me know if I am getting something wrong here.

Reply:
The original idea for this feature was the following:
-note that this assumes a process manager dividing the PEs at the halfway point across nodes
-note that this model wants to you launch the max PEs in the set and then reduce from that set

Thus the idea is to maintain that logical division. Its not the same idea as what I was doing with atomics where every PE is incast/active. In this case you reduce the active set (l/r stood for left and right).  You do however have to track the active set for later book keeping, but in your proposed example 3-7 is inactive and not target.

Reply:
Oh.. thanks for confirming this. I was thinking that `-l 3` means 3 sources and 13 targets. But, based on what you said, it is 3 sources and 8 targets and 5 unused/inactive. In that case, I see the reasoning why it should be based on `midpt`. 

But, with this model, although there is no need to, it is not possible to run a variable number of sources and targets at the same time. I am guessing there is no need to.

Reply:
my thought process was that if someone wanted less PEs on both sides then that person should just launch with less overall PEs.